geonode/upload/api/views.py

```diff
@@ -64,6 +66,51 @@ class UploadViewSet(DynamicModelViewSet):
     serializer_class = UploadSerializer
     pagination_class = GeoNodeApiPagination
 
+    def _emulate_client_upload_step(self, request, _step):
+        """Emulates the calls of a client to the upload flow.
+        It alters the content of the request object, so the same request should
+        be reused in the next call of this method.
+
+        Args:
+            request (Request): A request object with the query params given by the lasted step call.
+                               No params for the first call.
+            _step (string): The current step, used as an argument in the upload_view call.
+                            None for the first call.
+
+        Returns:
+            Response: response, upload_view response or a final response.
+            string: next_step, the next step to be performed.
+            boolean: is_final, True when the last step is executed or in case of errors.
+        """
+        response = upload_view(request, _step)
+        if response.status_code == status.HTTP_200_OK:
+            content = response.content
+            if isinstance(content, bytes):
+                content = content.decode('UTF-8')
+            data = json.loads(content)
+
+            required_input = data.get('required_input', None)
+            response_status = data.get('status', '')
+            response_success = data.get('success', False)
+            redirect_to = data.get('redirect_to', '')
+            if required_input or not response_success or not redirect_to or response_status == 'finished':
+                return response, None, True
+
+            # Prepare next step
+            next_step = redirect_to.split("?")[0].split(reverse("data_upload"))[1]
```

Comment:
Can we change this by using the `from urllib.parse import urlparse` or a `regex` in order to avoid all those very specific `splits`?
Just to be sure this is keep working with "whatever" `redirect_to` is passed, absolute or not.

Reply:
Good point, changing it.

Reply:
Updated.